Tests/SubstrataTests/EngineTests.swift

```diff
@@ -146,7 +146,7 @@ class EngineTests: XCTestCase {
         XCTAssertTrue(r is AnalyticsJS)
         
         let o = engine.object(key: "a")?.typed(JSObject.self)
-        XCTAssertNotNil(o)
+        XCTAssertNil(o)
```

Comment:
if AnalyticsJS is not a JSObject then what class is it? (inside JS)

Reply:
AnalyticsJS and JSObject are defined in Swift and exposed into the engine, but they do not have any relationship. One is not the super class of the other so the cast inside the `typed` function return `nil` (correctly)